domain/src/main/java/org/oppia/domain/profile/ProfileManagementController.kt

```diff
@@ -73,10 +225,114 @@ class ProfileManagementController @Inject constructor(
    * @param allowDownloadAccess New download access status for the profile being updated.
    * @return a [LiveData] that indicates the success/failure of this update operation.
    */
-  fun updateDownloadAccess(
+  @ExperimentalCoroutinesApi
+  fun updateAllowDownloadAccess(
     profileId: ProfileId, allowDownloadAccess: Boolean
   ): LiveData<AsyncResult<Any?>> {
-    return MutableLiveData(AsyncResult.success<Any?>(null))
+    val pendingLiveData = MutableLiveData(AsyncResult.pending<Any?>())
+    val deferred = profileDataStore.storeDataWithCustomChannelAsync(updateInMemoryCache = true) {
+      val profile = it.profilesMap[profileId.internalId] ?: return@storeDataWithCustomChannelAsync Pair(it, ProfileActionStatus.PROFILE_NOT_FOUND)
+      val updatedProfile = profile.toBuilder().setAllowDownloadAccess(allowDownloadAccess).build()
+      val profileDatabaseBuilder = it.toBuilder().putProfiles(profileId.internalId, updatedProfile)
+      Pair(profileDatabaseBuilder.build(), ProfileActionStatus.SUCCESS)
+    }
+    deferred.invokeOnCompletion {
+      if (it != null) {
+        logger.e("ProfileManagementController", "Failed to update allowDownloadAccess", it)
+        pendingLiveData.postValue(AsyncResult.failed(it))
+      } else {
+        when (deferred.getCompleted()) {
+          ProfileActionStatus.SUCCESS -> pendingLiveData.postValue(AsyncResult.success(null))
+          else -> pendingLiveData.postValue(AsyncResult.failed(ProfileNotFoundException("ProfileId ${profileId.internalId} does not match an existing Profile")))
+        }
+      }
+    }
+    return pendingLiveData
+  }
+
+  /**
+   * Login to the user's Profile by setting the current profile Id and updating profile's last logged in time.
+   *
+   * @param profileId the ID corresponding to the profile being logged into.
+   * @return a [LiveData] that indicates the success/failure of this login operation.
+   */
+  @ExperimentalCoroutinesApi
+  fun loginToProfile (profileId: ProfileId): LiveData<AsyncResult<Any?>> {
+    val pendingLiveData = MutableLiveData(AsyncResult.pending<Any?>())
+//    setCurrentProfileId(profileId).observeForever { setIdResult ->
```

Comment:
Why is this commented out? Can we remove it if it needs to be commented?

Reply:
Sorry, I think you reviewed my code while I was in the middle of addressing Ben's comments. Commented code was removed.